grip_api/src/grip_api/abstracts_widgets/abstract_socket.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright 2020, 2021, 2023 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+from abc import ABC, abstractmethod
+
+
+class AbstractSocket(ABC):
+
+    """
+        Abstract Socket class with set of variables and methods common to Socket and TerminalSocket
+    """
+
+    def __init__(self, socket_name, index=0, multi_connections=True):
+        """
+            Abstract widget initialisation
+
+            @param socket_name: Name of the socket (string)
+            @param index: Integer, stating the index of the socket on this side of the node
+            @param multi_connections: Boolean stating if the socket accepts multiple connections
+        """
+        # Store the id of the object
+        self.socket_id = id(self)
```

Comment:
Should be a property as children class do have a getter for this

Reply:
changed [here](https://github.com/shadow-robot/sr_grip/pull/16/commits/e8ac01c3bac38f78ed47b322574ebdca12c6bd2f)